src/ceph-volume/ceph_volume/tests/devices/lvm/strategies/test_bluestore.py

```diff
@@ -51,14 +52,15 @@ def test_device_is_lvm_member_fails(self, fakedevice, factory):
 
 class TestMixedType(object):
 
-    def test_filter_all_data_devs(self, fakedevice, factory):
+    def test_filter_all_data_devs(self, fakedevice, factory, monkeypatch):
         # in this scenario the user passed a already used device to be used for
         # data and an unused device to be used as db device.
         db_dev = fakedevice(used_by_ceph=False, is_lvm_member=False, rotational=False, sys_api=dict(size=6073740000))
         data_dev = fakedevice(used_by_ceph=True, is_lvm_member=False, rotational=True, sys_api=dict(size=6073740000))
         args = factory(filtered_devices=[data_dev], osds_per_device=1,
                        block_db_size=None, block_wal_size=None,
                        osd_ids=[])
+        monkeypatch.setattr(lvm, 'VolumeGroup', lambda x, **kw: [])
```

Comment:
have you tried with the `volume_groups` fixture?

Reply:
I haven't yet. Though I don't understand why the fixture does a `_purge` when the lvs call comes back empty anyway.

Reply:
the purge happens because it allows you to play around with the patched version of VolumeGroups like adding new volume groups with particular attributes